Makefile

```diff
@@ -0,0 +1,378 @@
+### CONFIG #
+#
+SHELL := bash # we want bash behaviour in all shell invocations
+PLATFORM := $(shell uname)
+platform := $(shell echo $(PLATFORM) | tr A-Z a-z)
+ifeq ($(PLATFORM),Darwin)
+platform_alt = macOS
+else
+platform_alt = $(platform)
+endif
+ARCH := $(shell uname -m)
+
+# https://stackoverflow.com/questions/4842424/list-of-ansi-color-escape-sequences
+RED := \033[0;31m
+GREEN := \033[0;32m
+YELLOW := \033[0;33m
+MAGENTA := \033[0;35m
+CYAN := \033[0;36m
+BOLD := \033[1m
+NORMAL := \033[0m
+INFO := $(CYAN)
+WARN := $(MAGENTA)
+
+XDG_CONFIG_HOME ?= $(CURDIR)/.config
+env::
+	@echo 'export XDG_CONFIG_HOME="$(XDG_CONFIG_HOME)"'
+KUBECONFIG_DIR = $(XDG_CONFIG_HOME)/kubectl
+KUBECONFIG ?= $(KUBECONFIG_DIR)/config
+$(KUBECONFIG_DIR):
+	mkdir -p $(@)
+env::
+	@echo 'export KUBECONFIG="$(KUBECONFIG)"'
+
+LOCAL_BIN := $(CURDIR)/bin
+$(LOCAL_BIN):
+	mkdir -p $@
+env::
+	@echo 'export PATH="$(LOCAL_BIN):$$PATH"'
+
+
+
+### DEPS #
+#
+ifeq ($(PLATFORM),Darwin)
+OPEN := open
+else
+OPEN := xdg-open
+endif
+
+GCLOUD_SDK_VERSION := 365.0.0
+GCLOUD_BIN := gcloud-$(GCLOUD_SDK_VERSION)-$(PLATFORM)-x86_64
+GCLOUD := $(LOCAL_BIN)/$(GCLOUD_BIN)
+GCLOUD_SDK_FILE := google-cloud-sdk-$(GCLOUD_SDK_VERSION)-$(PLATFORM)-x86_64.tar.gz
+GCLOUD_SDK_URL := https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/$(GCLOUD_SDK_FILE)
```

Comment:
I don't like downloading tools directly from the network. I have them installed on my machine, securely. Without checking for consistency (signs, checksums) you are introducing a vulnerability into Knative supply chain.

Reply:
I definitely think that this area should be improved.

I also think the zero-trust approach promoted by chainguard.dev is the right way to go, and considering the people that are behind this idea - @mattmoor @vaikas @n3wscott - I think that they would agree with this too.

Considering that we adopt the same approach in our existing GitHub Actions, and we suggest the same in the [Knative development guide](https://github.com/knative/eventing/blob/main/DEVELOPMENT.md), I don't see the approach in this PR being worse. Having said that, I'm glad that this is on your radar and I also think that we should close this gap as soon as possible, but not part of this PR.